shared/js/ui/models/whitelist.es6.js

```diff
@@ -21,6 +22,25 @@ Whitelist.prototype = window.$.extend({},
         value: false
       }
       })
+    },
+
+    addDomain: function (url) {
+      // we only whitelist domains, not full URLs
+      // and getDomain will return null if the URL is invalid
+      const domain = tldjs.getDomain(url)
```

Comment:
So basically the old method I was using, tldExists(), was not ok because it would return true for strings like "com" and "uk", in addition to valid domains.
I also looked into isValid() but it has the same issue - accepts random strings without tld at the end, like 'asdasd'.
This method, getDomain(), returns the domain if available, otherwise null, which is exactly what we want for validation. See docs [here](https://www.npmjs.com/package/tldjs#getdomain)

Reply:
The one problem is that this doesn't handle subdomains very well. E.g.:

1. Type in `andrey.duckduckgo.com` into the whitelist field
2. The entry that gets added is `duckduckgo.com`
(compare with going to `andrey.duckduckgo.com` and pressing the toggle to whitelist it, where the full `andrey.duckduckgo.com` is added)

Reply:
Good find! I can prefix it using getSubDomain - if there's no subdomain it just returns an empty string.